vspace/src/main/java/edu/asu/diging/vspace/core/services/ISpaceManager.java

```diff
@@ -28,5 +29,7 @@
     void deleteSpaceById(String id) throws SpaceDoesNotExistException;
     
     List<SpaceLink> getIncomingLinks(String id);
+    
+    Iterable<Space> addIncomingLinkInfoToAllSpaces(Iterable<Space> allSpaces);
```

Comment:
still the same issue, is it all spaces or the passed spaces (which don't have to be all of them)?

Reply:
I am passing all the spaces to this method. That's why I named it allSpaces.